galasa-managers-parent/galasa-managers-cloud-parent/dev.galasa.docker.manager/src/main/java/dev/galasa/docker/internal/DockerContainerResourceMonitor.java

```diff
@@ -0,0 +1,148 @@
+     * @param containers
+     * @param client
+     */
+    private void killContainers(List<String> containers, IHttpClient client) {
+        logger.info("Shutting down oprhaned containers");
+        try {
+            for (String id : containers) {
+                HttpClientResponse<String> resp = client.deleteText("/containers/"+id+"?force=true");
+                if (resp.getStatusCode() != 204) {
+                    logger.error("Something went wrong when removing container: " + resp.getStatusLine());
+                    return;
+                }
+            }
+        } catch (HttpClientException e) {
+            logger.error("Failed to kill containers.", e);
+        }        
+    }
+
+    /**
+     * Looks at all containers on a Engine, locates Galasa specifics and ensures they have a decicated slot
+     * @param engine
+     * @param client
+     * @return
+     */
+    private List<String> getOrphanedContainers(String engine, IHttpClient client) {
+        List<String> orphanedContainers = new ArrayList<>();
+        try {
+            HttpClientResponse<String> resp = client.getText("/containers/json");
+            if (resp.getStatusCode() != 200) {
+                logger.error("Something went wrong when retrieving containers: " + resp.getStatusLine());
+                return orphanedContainers;
+            }
+
+            DockerContainerJSON[] activeContainers = gson.fromJson(resp.getContent(), DockerContainerJSON[].class);
+            for (DockerContainerJSON container : activeContainers) {
+                String runName = container.getLabels().getRunId();
+                String slotId = container.getLabels().getSlotId();
+                // Other non Galasa pod
+                if (runName == null) {
+                    continue;
+                }
+
+                // Check Slot name against runID. If Null or another run then container orphaned
+                if (!runName.equals(dss.get("engine."+engine+".slot."+slotId))) {
+                    orphanedContainers.add(container.getId());
+                }
+            }
+        } catch (HttpClientException | FrameworkException e) {
+            logger.error("Failed to get containers.", e);
+        }
+        return orphanedContainers;
+    }
+
+    /**
+     * Looks at CPS for the docker Engines available
+     * @return
+     */
+    private Map<String, IHttpClient> getDockerEngines() {
+        HashMap<String, IHttpClient> engines = new HashMap<>();
+        try { 
+            // This will have to be changed if we support engine clusters
+            String[] enginesTags = cps.getProperty("default", "engines").split(",");
+            for (String engine : enginesTags) {
+                String hostname = cps.getProperty("engine", "hostname", engine);
+                String port = cps.getProperty("engine", "port", engine);
+
+                IHttpClient client = StandAloneHttpClient.getHttpClient(3600, logger);
```

Comment:
shouldnt you check to see if you have already got a Ihttpclient for this engine/hostname/port before creating a new one.

Reply:
needs to be a little sophisticated than this.    to avoid memory leaks and over usage as this cleanup will be sharing the server with other clean ups